source/shadow.c

```diff
@@ -454,20 +470,13 @@ static ShadowStatus_t extractShadowRootAndName( const char * pTopic,
         /* Extract shadow name. */
         shadowStatus = validateName( &( pTopic[ *pConsumedTopicLength ] ),
                                      topicLength - *pConsumedTopicLength,
+                                     SHADOW_NAME_MAX_LENGTH,
                                      pShadowNameLength );
 
         if( shadowStatus != SHADOW_SUCCESS )
         {
             shadowStatus = SHADOW_SHADOWNAME_PARSE_FAILED;
-            LogDebug( ( "Not related to Shadow. Failed to parse shadow name in pTopic %s.", pTopic ) );
-        }
-        else if( *pShadowNameLength > SHADOW_NAME_MAX_LENGTH )
-        {
-            shadowStatus = SHADOW_SHADOWNAME_PARSE_FAILED;
-            LogDebug( ( "Not related to Shadow. Shadow name length of %u exceeds maximum allowed length %u in pTopic %s.",
-                        ( unsigned int ) *pShadowNameLength,
-                        ( unsigned int ) SHADOW_NAME_MAX_LENGTH,
-                        pTopic ) );
+            LogDebug( ( "Not a Shadow topic. Failed to parse shadow name in pTopic %s.", pTopic ) );
```

Comment:
nit: this error message is confusing – `validateName()` has already performed all validation and logging, so this message just takes up space without adding value.

Reply:
Thanks. I have thought about it too. The value addition I see is the topic name is printed out in logs. `validateName()` doesn't have access to the complete topic string. However, application will be passing the topic string. Since the library is printing logs to show exact reason why it failed to identify a shadow topic, I think that will be enough. I will go ahead and remove it.
About the space, The string will be taking space in the binary only if debug logs are enabled for the library though.